src/main/java/ch/njol/skript/effects/EffFireworkLaunch.java

```diff
@@ -72,6 +76,7 @@ protected void execute(Event e) {
 			meta.addEffects(effects.getArray(e));
```

Comment:
While here, can you extract the getArray out of the loop to avoid double evaluation